src/pw/pw_methods.F

```diff
@@ -1376,6 +1381,30 @@ SUBROUTINE fft_wrap_pw1pw2(pw1, pw2, debug)
          CASE ("FW_R3DC1D")
 #if defined (__PW_CUDA)
             CALL pw_cuda_r3dc1d_3d(pw1, pw2, scale=norm)
+#elif defined (__PW_FPGA)
+            ! check if bitstream for the fft size is present
+            ! if not, perform fft3d in CPU
+            IF (pw_fpga_check_bitstream(n) .EQV. .TRUE.) THEN
+               ALLOCATE (c_out(n(1), n(2), n(3)))
+
+               CALL copy_rc(pw1%cr3d, c_out)
+#if defined (__PW_FPGA_SP)
+               CALL pw_fpga_r3dc1d_3d_sp(n, c_out)
+#else
+               CALL pw_fpga_r3dc1d_3d_dp(n, c_out)
+#endif
+               CALL zdscal(n(1)*n(2)*n(3), norm, c_out, 1)
+
+               CALL pw_gather(pw2, c_out)
+
+               DEALLOCATE (c_out)
+            ELSE
+               ALLOCATE (c_out(n(1), n(2), n(3)))
```

Comment:
possibly move the `ALLOCATE(c_out..)/DEALLOCATE` outside of the `IF/ELSE`

Reply:
same below

Reply:
This I should've noticed.. Thanks :)